rosplan_planning_system/src/PlannerInterface/LPGPlannerInterface.cpp

```diff
@@ -64,7 +64,7 @@ namespace KCL_rosplan {
 		if(dit!=std::string::npos) str.replace(dit,6,domain_path);
 		std::size_t pit = str.find("PROBLEM");
 		if(pit!=std::string::npos) str.replace(pit,7,problem_path);
-        std::string updatePlan = "cp "+data_path+"../../rosplan_planning_system/common/lpgplan.SOL"+" "+data_path+"plan.pddl";
+        std::string updatePlan = "cp "+data_path+"lpgplan.SOL"+" "+data_path+"plan.pddl";
```

Comment:
Does this require that LPG creates the solution file in the data path? Or to put it another way, the data path parmaeter must point towards LPG's working directory, or the copy will fail.

Reply:
It is not mandatory to point the data path to the LPG working directory. The issue is that file plan created by LPG has either no extension or the .SOL extension, we can not export the plan directly to a file called "plan.pddl". I opted for that particular path as to keep both the .SOL file and .pddl file in the same place, for better user experience.

Reply:
This line is setting up a command "cp [data_path]/lpgplan.SOL [data_path]/plan.pddl"

Are you saying that I can set the data_path launch parameter to any directory, and the lpgplan.SOL file will be there?

Reply:
The *.SOL file is the default output of the LGP planner. When you run the LGP planner, you need to specifiy the name of the output file name and path (ex: -out lgpplan). The final output file is the selected name + .SOL. The file can be created in any directory. For better clarification about his I recommend to run the LPG planner with a demo domain/problem. 

I am thinking that maybe a commented planner_command with LPG in the main launch file would help.

Reply:
I think this is what I was getting at:
The path (and name) of the *.SOL file is set in the planner command parameter, but the hard-coded copy command uses the data path parameter. There is no hint to the user that these paths should be the same.
A better solution would be to alter the copy command so that it automatically uses the path in which the SOL file is created (either from the -out argument, or whatever is the default if the user does not specify -out).
If this is not possible, using ROS_INFO messages to describe to the user would also help. From the user perspective, I will not understand what is happening if the "cp" fails, so some output like this might help:
"Running planner command..."
"Creating SOL file in directory: ..."
"Copying SOL file from (data_path) [path] to [path]"
With the above, I might be able to diagnose what went wrong.

Reply:
Modified the code so that the -out path typed by the user is changed to 'data_path/lpgplan'.

Added a ROS_INFO which informs the user of the change and instructs to have -out as the last parameter in planner_command.